lib/service_pub_queue_rabbit.js

```diff
@@ -72,7 +74,28 @@ class ServicePubQueueRabbit extends Component.mixin(AsyncEmitter) {
 
         this._channels[service][event] = channel;
 
-        yield channel.assertQueue(queueName, { durable: true });
+        const queueOpts = {
+            durable: true,
+        };
+
+        if (this._withExpire) {
+            Object.assign(queueOpts, { expires: this.expireTime });
+        }
+
+        // by default, assert queue with `expires` header. if queue is exists and
+        // don't have `expires` header, rabbitMQ will throw an error and close channel.
+        // we catch the error on `close` or `error` emit (amqp.node lib emit those error)
+        // and re-init this class
+        try {
+            yield channel.assertQueue(queueName, queueOpts);
+        } catch (e) {
+            this._initializing = false;
+
+            // dont use `expires` header on re-init
+            this._withExpire = false;
+            this._connection.on("close", this.init.bind(this));
+            this._connection.on("error", this.init.bind(this));
```

Comment:
this is the fallback mechanism if the connection emit `close` or `error` (this is coming from the *amqp.node* if the `assertQueue` throws error)